docs/sdks/typescript/index.md

```diff
@@ -55,927 +67,740 @@ Each SpacetimeDB client depends on some bindings specific to your module. Create
+##### Callback `onError`
 
-#### Example
+```typescript
+SubscriptionBuilder.onError(
+    callback: (ctx: ErrorContext, error: Error) => void
+): SubscriptionBuilder
 
-```ts
-spacetimeDBClient.onConnect((token, identity, address) => {
-  console.log('Connected to SpacetimeDB');
-  console.log('Token', token);
-  console.log('Identity', identity);
-  console.log('Address', address);
-});
 ```
 
----
+Register a callback to run if the subscription is rejected or unexpectedly terminated by the server. This is most frequently caused by passing an invalid query to [`subscribe`](#method-subscribe).
 
-### `SpacetimeDBClient` onError
 
-Register a callback to be invoked upon an error.
+##### Method `subscribe`
 
-```ts
-onError(callback: (...args: any[]) => void): void
+```typescript
+SubscriptionBuilder.subscribe(queries: string | string[]): SubscriptionHandle
 ```
 
-#### Parameters
+Subscribe to a set of queries.
 
-| Name       | Type                           |
-| :--------- | :----------------------------- |
-| `callback` | (...`args`: `any`[]) => `void` |
+See [the SpacetimeDB SQL Reference](/docs/sql#subscriptions) for information on the queries SpacetimeDB supports as subscriptions.
 
-#### Example
+##### Method `subscribeToAllTables`
 
-```ts
-spacetimeDBClient.onError((...args: any[]) => {
-  console.error('ERROR', args);
-});
+```typescript
+SubscriptionBuilder.subscribeToAllTables(): void
 ```
 
-### Class `Identity`
-
-A unique public identifier for a user of a database.
-
-Defined in [spacetimedb-sdk.identity](https://github.com/clockworklabs/spacetimedb-typescript-sdk/blob/main/src/identity.ts):
-
-| Constructors                                    | Description                                  |
-| ----------------------------------------------- | -------------------------------------------- |
-| [`Identity.constructor`](#identity-constructor) | Creates a new `Identity`.                    |
-| Methods                                         |                                              |
-| [`Identity.isEqual`](#identity-isequal)         | Compare two identities for equality.         |
-| [`Identity.toHexString`](#identity-tohexstring) | Print the identity as a hexadecimal string.  |
-| Static methods                                  |                                              |
-| [`Identity.fromString`](#identity-fromstring)   | Parse an Identity from a hexadecimal string. |
+Subscribe to all rows from all public tables. This method is provided as a convenience for simple clients. The subscription initiated by `subscribeToAllTables ` cannot be canceled after it is initiated. You should [`subscribe` to specific queries](#method-subscribe) if you need fine-grained control over the lifecycle of your subscriptions.
 
-## Constructors
+#### Type `SubscriptionHandle`
 
-### `Identity` constructor
-
-```ts
-new Identity(data: Uint8Array)
+```rust
+module_bindings::SubscriptionHandle
```

Comment:
There is technically no such type in the TypeScript SDK. We just return: `SubscriptionHandleImpl<DBView, Reducers, SetReducerFlags>`

Reply:
Not a huge deal though since it is that type in spirit.

Reply:
Does this mean that a user who wants to store a subscription handle in some sort of data structure has to either write out `SubscriptionHandleImpl<DBView, Reducers, SetReducerFlags>` or cast to `any`?